interfaces/repo.go

```diff
@@ -609,6 +609,38 @@ func (r *Repository) securitySnippetsForSnap(snapName string, securitySystem Sec
 	return snippets, nil
 }
 
+// RecordInterfacesAffectingSnap records all the security changes of a given snap.
+func (r *Repository) RecordInterfacesAffectingSnap(snapName string, rec Recorder) error {
```

Comment:
If we pass the `Specification` directly we don't need to keep track of all the possible security systems in the repository. If we go ahead with this I think we will also need something like 

`AddSecuritySystem(securitySystem SecuritySystem, func makeSpec() Specification) error`